core-concepts/modules/ROOT/pages/drivers/analyze.adoc

```diff
@@ -0,0 +1,296 @@
+= Analyzing queries
+:pageTitle: Analyzing queries
+:Summary: Compiling & analyzing TypeDB queries.
+:keywords: typedb, driver, analyze, type-inference
+:test-python: true
+
+TypeDB allows the user to "analyze" a query without having it execute against the data.
+During analysis, a query is parsed to the internal representation and type-checked -
+allowing the user to check their query for syntax and typing errors against a schema.
+The envisioned use is to facilitate developer tooling around TypeDB, such as
+plugins to validate TypeQL queries when application code is being compiled,
+and debugging type errors.
```

Comment:
let's throw in some kind of AI reference: for AI query generators to validate queries automatically.